Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromCondDB.cc

```diff
@@ -121,20 +123,89 @@ GEMGeometry* GEMGeometryBuilderFromCondDB::build(const RecoIdealGeometry& rgeo)
     }
 
     ReferenceCountingPointer<BoundPlane> surf(bp);
+    // Create the superchamber
+    if(chid.layer()==1) {
+      GEMDetId schid(chid.region(),chid.ring(),chid.station(),0,chid.chamber(),0);
+      GEMSuperChamber* sch = new GEMSuperChamber(schid,surf);
+      LogDebug("GEMGeometryBuilder")<<"GEM SuperChamber created with id = "<<schid<<" and added to the GEMGeometry"<<std::endl;
+      geometry->add(sch);
+    }
     // Create the chamber 
     GEMChamber* ch = new GEMChamber (chid, surf); 
+    LogDebug("GEMGeometryBuilder")<<"GEM Chamber created with id = "<<chid<<" = "<<chid.rawId()<<" and added to the GEMGeometry"<<std::endl;
+    LogDebug("GEMGeometryBuilder")<<"GEM Chamber has following eta partitions associated: "<<std::endl;
     // Add the etaps to rhe chamber
     for(std::list<GEMEtaPartition *>::iterator gepl=gepls.begin();
     gepl!=gepls.end(); gepl++){
       ch->add(*gepl);
+      LogDebug("GEMGeometryBuilder")<<"   --> GEM Eta Partition "<<GEMDetId((*gepl)->id())<<std::endl;
     }
     // Add the chamber to the geometry
     geometry->add(ch);
 
   }
-  */
+  
+  // The super chamber is composed of 2 chambers.
+  // It's detId is layer 0, chambers are layer 1 and 2
+
+  auto& superChambers( geometry->superChambers());
+  // construct the regions, stations and rings. 
+  for( int re = -1; re <= 1; re = re+2 ) {
+    GEMRegion* region = new GEMRegion( re );
+    for( int st = 1; st <= GEMDetId::maxStationId; ++st ) {
+      GEMStation* station = new GEMStation(re, st);
+      std::string sign( re==-1 ? "-" : "");
+      std::string name("GE" + sign + std::to_string(st) + "/1");
+      station->setName(name);
+      for( int ri = 1; ri <= 1; ++ri ) {
+	GEMRing* ring = new GEMRing( re, st, ri );
+	for( unsigned sch = 0; sch < superChambers.size(); ++sch ) {
+	  GEMSuperChamber* superChamber = const_cast<GEMSuperChamber*>( superChambers.at(sch));
```

Comment:
@ianna and i see a simple solution - instead of making the super chambers, giving them to the geometry and then getting them back (as const) from the geometry)

make the super chambers, give them to the geometry, but keep a local vector of them for use in this loop. then it would be non-const